test/models/fact_check_test.rb

```diff
@@ -741,4 +741,65 @@ def setup
     User.current = nil
     assert_equal u, fc.author
   end
+
+  test "should assign default channel 'manual' for a regular user" do
+    fc = create_fact_check
+    assert_equal 'manual', fc.channel
+  end
+
+  # this was the only way so far I got this test to work so we could set User.current
+  test "should assign default channel 'api' for a BotUser" do
```

Comment:
@caiosba test using `User.current`.

Reply:
Another reason for the `set_channel` callback to rely only on `self.user` - the test can also only rely on `self.user`.